torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2818,7 +2820,7 @@ def _pre_forward_unshard(
             self._unshard(handles)
             handles_key = tuple(handles)
             self._needs_pre_forward_unshard[handles_key] = False
-            torch.cuda.current_stream().wait_stream(self._streams["unshard"])
+            self._streams["computation"].wait_stream(self._streams["unshard"])
```

Comment:
same question, changing from current stream to a previously saved current stream might change behavior, right?